### ISSUE TITLE
Native Image building fails with JUnit Platform 5.9.0

### ISSUE THREAD

**Issue by melix** [graalvm/native-build-tools#300]
Building a Micronaut application which uses JUnit Platform 5.9.0 will fail with the following error:

```
Error: java.util.concurrent.ExecutionException: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: No instances of org.junit.platform.reporting.open.xml.OpenTestReportGeneratingListener are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=org.junit.platform.reporting.open.xml.OpenTestReportGeneratingListener.
```

A hotfix was deployed in Micronaut Test: https://github.com/micronaut-projects/micronaut-test/pull/618/files#diff-020ea83d544d7fe86f8cc9bdbcf3eba4bd2d37bb3ed842d10eae76b4bf9a3c6e

But a "real" fix needs to be implemented in `JUnitPlatformFeature`.

**Comment by sbrannen** [comment]
Is this due to `OpenTestReportGeneratingListener.NAMESPACE_REGISTRY` being a `static final` field?

If so, we (the JUnit team) could instantiate the `NamespaceRegistry` lazily (on demand).

In addition, `OpenTestReportGeneratingListener` is an _optional_ listener and has to be manually enabled by setting the `junit.platform.reporting.open.xml.enabled` property to `true`. So I wonder what is enabling that. Maybe recent versions of Gradle automatically enable that?

/cc @marcphilipp

**Comment by marcphilipp** [comment]
@melix @gradinac Should we fix this in JUnit 5.9.1 as suggested by Sam?

**Comment by sbrannen** [comment]
> In addition, `OpenTestReportGeneratingListener` is an _optional_ listener and has to be manually enabled by setting the `junit.platform.reporting.open.xml.enabled` property to `true`. So I wonder what is enabling that. Maybe recent versions of Gradle automatically enable that?

Never mind: I'd forgotten that `OpenTestReportGeneratingListener` will always be loaded due to the `ServiceLoader` mechanism. It's only "optional" in the sense that it's a no-op if it's not explicitly _enabled_.

**Comment by sbrannen** [comment]
This will be addressed in JUnit 5.9.1:

- https://github.com/junit-team/junit5/issues/3035

**Comment by sbrannen** [comment]
@melix and @gradinac, when are you planning on upgrading to JUnit 5.9.x?

**Comment by melix** [comment]
Micronaut itself already has upgraded to 5.9.0

**Comment by sbrannen** [comment]
Sorry, I was unclear. I meant Native Build Tools when I said "you".

In other words, I'm asking when you plan to upgrade NBT to JUnit 5.9.x.

**Comment by sbrannen** [comment]
> But a "real" fix needs to be implemented in `JUnitPlatformFeature`.

If you choose to add a fix for JUnit 5.9.0, that should go in `PlatformConfigProvider`.

**Comment by sbrannen** [comment]
To fix this bug, add `"org.junit.platform.reporting.open.xml.OpenTestReportGeneratingListener"` to the `buildTimeInitializedClasses` string array in `org.graalvm.junit.platform.config.platform.PlatformConfigProvider.onLoad(NativeImageConfiguration)`.

I have tested this with the following combinations.

----

## Native Build Tools 0.9.13

✅ NBT 0.9.13 :: JUnit 5.8.1
✅ NBT 0.9.13 :: JUnit 5.8.2
❌ NBT 0.9.13 :: JUnit 5.9.0
✅ NBT 0.9.13 :: JUnit 5.9.1-SNAPSHOT

----

## Native Build Tools 0.9.14-SNAPSHOT -- without fix

✅ NBT 0.9.14-SNAPSHOT :: JUnit 5.8.1
✅ NBT 0.9.14-SNAPSHOT :: JUnit 5.8.2
❌ NBT 0.9.14-SNAPSHOT :: JUnit 5.9.0
✅ NBT 0.9.14-SNAPSHOT :: JUnit 5.9.1-SNAPSHOT

----

## Native Build Tools 0.9.14-SNAPSHOT -- with fix

✅ NBT 0.9.14-SNAPSHOT :: JUnit 5.8.1
✅ NBT 0.9.14-SNAPSHOT :: JUnit 5.8.2
✅ NBT 0.9.14-SNAPSHOT :: JUnit 5.9.0
✅ NBT 0.9.14-SNAPSHOT :: JUnit 5.9.1-SNAPSHOT

----
